File: 2020/3.rb
lines = File.open("3.txt").readlines.map(&:strip)
count_overall = 1
(1..7).step(2) do |step|
	count = 0
	idx = 0
	lines.each do |x|
		count += 1 if x[idx % x.size] == "#"
		idx += step
	end
	puts count if step == 3
	count_overall *= count
end

count = 0
idx = 0
(0..lines.size).step(2) do |x|
	count += 1 if lines[x][idx % lines[x].size] == "#"
	idx += 1
end
count_overall *= count

puts count_overall

File: 2020/3.py
grid = list(map(str.strip, open("3.txt").readlines()))

r, c, count = 0, 0, 0
while r < len(grid):
    if grid[r][c] == "#": count += 1
    c = (c + 3) % len(grid[0])
    r += 1

# part 1: No issues
print (count)

product = 1
for i in [1, 3, 5, 7]:
    r, c, count = 0, 0, 0
    while r < len(grid):
        if grid[r][c] == "#": count += 1
        c = (c + i) % len(grid[0])
        r += 1
    product *= count

r, c, count = 0, 0, 0
while r < len(grid):
    if grid[r][c] == "#": count += 1
    r = r + 2
    c = (c + 1) % len(grid[0])
product *= count

# part 2: Initially did not realize that there is one that goes down by two, which slowed me down
print (product)
